### PR TITLE
 chore/all: stub handlers for client->source elders RPCs 

### PR DESCRIPTION
**Pull request by Fraser999**

### CODE REVIEW

File: src/coins_handler.rs [hunk 1]
```diff
@@ -8,18 +8,29 @@
...
+use safe_nd::NodePublicId;
+use std::{
+    fmt::{self, Display, Formatter},
+    path::Path,
+};
 
 const COINS_DB_NAME: &str = "coins.db";
 
 pub(crate) struct CoinsHandler {
+    _id: NodePublicId,
```
Comment (octol) [inline src/coins_handler.rs]: Nitpick: you're using this in `fmt` so you can remove the underscore

File: src/source_elder.rs [hunk 2]
```diff
@@ -147,11 +175,145 @@ impl SourceElder {
...
+
+    fn is_valid_client_signature(
+        &self,
+        client_id: &PublicId,
+        request: &Request,
+        message_id: &MessageId,
+        signature: &Signature,
+    ) -> bool {
+        let pub_key = match client_id {
+            PublicId::Node(_) => return false, // unreachable - no need to log
```
Comment (octol) [inline src/source_elder.rs]: Should we put in a `warn!` here to detect if this (accidentally) becomes reachable?

File: src/vault.rs [hunk 3]
```diff
@@ -116,14 +132,48 @@ impl Vault {
...
+                info!("{}: Unexpected event: {}", self, event);
             }
         }
         None
     }
 
-    fn handle_action(&mut self, _action: Action) -> Option<Action> {
-        None
+    fn handle_action(&mut self, action: Action) -> Option<Action> {
+        use Action::*;
```
Comment (octol) [inline src/vault.rs]: I must say I like this habit of importing only in local scope, something I'll do more now :)

File: src/source_elder.rs (hunk 2 above)
Comment (Fraser999) [reply src/source_elder.rs]: Yeah - fair point.  I'll make it `error!` though, since it'd represent a true logic error.

File: src/vault.rs (hunk 3 above)
Comment (Fraser999) [reply src/vault.rs]: Well, afaik, we're only really meant to do this in fairly limited cases.  Normally we're supposed to do all the importing at the start of the module.  We agreed to make an exception for this sort of case where repeating the enum's name for every variant is a bit tedious and doesn't improve clarity.

Comment (octol) [reply src/vault.rs]: Ok fair enough if it's style rule

(As a general overarching rule though, isn't it preferable to only keep things in the smallest scope necessary?)

Comment (Fraser999) [reply src/vault.rs]: Yeah, I can certainly see that point too, although what we're doing seems to be fairly standard.  One for the next coding standards discussion maybe?

**Pull request merged**